pkg/apiserver/resthandler.go

```diff
@@ -82,7 +82,13 @@ func GetResource(r RESTGetter, ctxFn ContextFunc, namer ScopeNamer, codec runtim
 }
 
 func parseSelectorQueryParams(query url.Values, version, apiResource string) (label, field labels.Selector, err error) {
-	label, err = labels.Parse(query.Get("labels"))
+	var labelString string
+	if version == "v1beta1" || version == "v1beta2" {
+		labelString = query.Get("labels")
+	} else {
+		labelString = query.Get("label-selector")
```

Comment:
This kind of version test is not acceptable. Among other problems, it would break Openshift. It needs to be abstracted away. I don't think that parameterizing a function by version is the right thing, either.

@smarterclayton Which interface needs to be extended?